src/components/button/button.test.ts

```diff
@@ -0,0 +1,70 @@
+import path from 'path';
+import type { ElementHandle, PleasantestUtils } from 'pleasantest';
+import { withBrowser } from 'pleasantest';
+import { loadTwigTemplate } from '../../../test-utils';
+
+const buttonMarkup = loadTwigTemplate(path.join(__dirname, './button.twig'));
+// Helper to test the JS-driven toggling button
+const initTogglingButtonJS = (
+  utils: PleasantestUtils,
+  togglingButton: ElementHandle
+) =>
+  utils.runJS(
+    `
+    import { initTogglingButton } from './toggling-button'
+    export default (togglingButton) => initTogglingButton(togglingButton)
+    `,
+    [togglingButton]
+  );
+
+test(
+  'should render a button element by default',
+  withBrowser(async ({ utils, screen }) => {
+    // If I don't pass _something_ to buttonMarkup(), I get an error.
+    // To get around that, I pass an empty object.
+    await utils.injectHTML(await buttonMarkup({}));
+
+    const button = await screen.getByRole('button');
+    await expect(button).toBeVisible();
+  })
+);
+
+test(
+  'should render an anchor element when href is provided',
+  withBrowser(async ({ utils, screen }) => {
+    await utils.injectHTML(await buttonMarkup({ href: '#' }));
+
+    const linkButton = await screen.getByRole('link');
+    await expect(linkButton).toBeVisible();
+  })
+);
+
+test(
+  'should render a button when aria_pressed is set',
+  withBrowser(async ({ utils, screen }) => {
+    await utils.injectHTML(
+      await buttonMarkup({ tag_name: 'a', aria_pressed: 'false' })
+    );
+
+    const button = await screen.getByRole('button');
+    await expect(button).toBeVisible();
```

Comment:
@gerardo-rodriguez I'm wondering whether the visibility check is really needed here (and in the other tests). `getByRole` acts as its own assertion, since it will throw if no matching element is found, so I don't think the visibility check is necessary.

Maybe another assertion that would be helpful is to check that the button has the right text passed through using `expect(button).toHaveTextContent(/.../)`?

Reply:
Thanks, @calebeby, appreciate the feedback!

Reply:
Done! https://github.com/cloudfour/cloudfour.com-patterns/pull/1318/commits/b66277c3db90e23bbef3b1b3085194c1f24f8d7d